### PR TITLE
feat: Exposed LLMWhisperer options - extraction_mode, ocr_mode

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What
- Exposed arguments for LLMWhisperer - `extraction_mode` and `ocr_mode`
- Since its a breaking change (removal of the previously supported `mode`) - I've bumped the adapter version to 0.3.0

## Why
- These parameters help control the text extraction by LLMWhisperer and it needs to be configurable by the user

## How
- These params are read when the adapter is first created - and will be a part of the `self.config` object  of the adapter

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

-

## Notes on Testing
- Created an adapter with different modes and tried to index on prompt studio

## Screenshots
image
...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/adapters/x2text/llm_whisperer/src/llm_whisperer.py [hunk 1]
```diff
@@ -72,18 +75,15 @@ def __make_request(
         files = None
```
Comment (muhammad-ali-e) [inline src/unstract/adapters/x2text/llm_whisperer/src/llm_whisperer.py:75]: I think, Without files params/ or None we don't need to call LLM whisperer .Insetad we can raise some error from here self. Don't want in this PR. we can make it later also filne.

File: src/unstract/adapters/x2text/llm_whisperer/src/static/json_schema.json [hunk 2]
```diff
@@ -12,13 +12,26 @@
...
       "format": "uri",
       "default": "http://unstract-llm-whisperer:3005",
       "description": "Provide the URL of the LLM Whisperer server."
+    },
+    "extraction_mode": {
+      "type": "string",
+      "title": "Extraction Mode",
+      "enum": ["text", "ocr"],
+      "default": "text",
+      "description": "Text mode tries to extract text from PDF and falls to OCR if the PDF is a scanned image PDF. This should be your default selection. Use OCR mode if you want to force OCR to extract text. This could be useful if you are dealing with malformed PDFs."
```
Comment (muhammad-ali-e) [inline src/unstract/adapters/x2text/llm_whisperer/src/static/json_schema.json:27]: Did you check, How this diplayes in UI. If this descption larger maay some brakes in UI .

**Review by muhammad-ali-e — APPROVED** [review]
LGTM - left some comments

File: src/unstract/adapters/x2text/llm_whisperer/src/static/json_schema.json (hunk 2 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/adapters/x2text/llm_whisperer/src/static/json_schema.json:27]: It does render fine
image

**Review by nehabagdia — APPROVED** [review]

**Review by gaya3-vijayakumar — COMMENTED** [review]
Looks good

**Pull request merged**